class.c

```diff
@@ -795,6 +795,16 @@ Perl_class_seal_stash(pTHX_ HV *stash)
 
         aux->xhv_class_initfields_cv = initfields;
     }
+    else {
+        /* we had errors, clean up and don't populate initfields */
+        PADNAMELIST *fieldnames = aux->xhv_class_fields;
+        for(SSize_t i = 0; fieldnames && i <= PadnamelistMAX(fieldnames); i++) {
```

Comment:
One small thought: usually cleanup happens from the end backwards; so maybe

```c
for(SSize_t i = PadnamelistMAX(fieldnames); i >= 0; i--)
```

would be better here?

Reply:
That seems reasonable, updated.